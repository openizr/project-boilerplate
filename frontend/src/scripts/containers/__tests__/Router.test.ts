/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/svelte';
import Router from 'scripts/containers/Router.svelte';

jest.mock('scripts/store/routes', () => ({
  default: {
    // eslint-disable-next-line @typescript-eslint/no-var-requires, global-require
    '/': (): Any => Promise.resolve(require('scripts/pages/Home.svelte')),
  },
}));

jest.mock('scripts/locale/index', () => ({ default: jest.fn(() => Promise.resolve({ LABEL_TEST: 'Test' })) }));

jest.mock('scripts/store/index', () => ({
  mutate: jest.fn(),
  dispatch: jest.fn(),
  useCombiner: jest.fn(() => ({
    subscribe: (callback: Any): Any => {
      callback({ route: (process.env.NOT_FOUND === 'true' ? null : '/') });
      return jest.fn();
    },
  })),
}));

describe('containers/Router', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    delete process.env.LOADING;
    delete process.env.NOT_FOUND;
  });

  test('renders correctly - loading page', () => {
    const { container } = render(Router);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - page found', async () => {
    const { container } = render(Router);
    await new Promise((resolve) => { setTimeout(resolve); });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - page not found', async () => {
    process.env.NOT_FOUND = 'true';
    const { container } = render(Router);
    await new Promise((resolve) => { setTimeout(resolve); });
    expect(container.firstChild).toMatchSnapshot();
  });
});
