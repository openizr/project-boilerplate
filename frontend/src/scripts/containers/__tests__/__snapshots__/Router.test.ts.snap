// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Router renders correctly - loading page 1`] = `
<div>
  <div>
    Loading...
  </div>
  
</div>
`;

exports[`containers/Router renders correctly - page found 1`] = `
<div>
  <p>
    <img
      alt="test"
      class="svelte-1int8dr"
    />
     
    <img
      alt="test"
      class="svelte-1int8dr"
    />
     
    Test
  </p>
  
  
</div>
`;

exports[`containers/Router renders correctly - page not found 1`] = `
<div>
  <div>
    Not Found.
    object
  </div>
  
</div>
`;
